Client/control/mouse.py

```
@@ -0,0 +1,47 @@
+import pyautogui
+import logging
+
+# Disable PyAutoGUI fail-safe for this application as the LLM needs control
+# But we should be careful. Actually, let's keep it enabled but warn.
+pyautogui.FAILSAFE = True
+
+def validate_coordinates(x, y):
+    screen_width, screen_height = pyautogui.size()
+    if not (0 <= x <= screen_width and 0 <= y <= screen_height):
+        raise ValueError(f"Coordinates ({x}, {y}) are out of bounds for screen size {screen_width}x{screen_height}")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)